lib/reverse_sentence.rb

```diff
@@ -1,6 +1,40 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n + n/2 + n + n/2) = O(n)
+  # it basically goes n times with n being how long the 
+  # sentence is (in letters)
+# Space complexity: O(left and right and regex and reversed words) = O(n)
+  # it will need to create a left and right and regex variable each time that 
+  # won't change depending on input, but the list of words will increase by n 
+  # if n increases (in number of words)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return if my_sentence == nil || my_sentence.length <= 1
+  # loop through my_sentence, reversing it from last to first
+  left = 0
+  right = my_sentence.length - 1
+  until left == right || right < left
+    my_sentence[left], my_sentence[right] = my_sentence[right], my_sentence[left]
+    left += 1
+    right -= 1
+  end
```

Comment:
Question, since you reverse things twice in this method could you dry things up a bit by making a helper method?